dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/PolicyExecutorImpl.java

```diff
@@ -959,4 +960,43 @@ public void shutdown() {
         enqueue(policyTaskFuture, maxWaitForEnqueueNS.get(), null);
         return policyTaskFuture;
     }
+
+    public void introspect(PrintWriter out) {
+        final String INDENT = "  ";
+        final String DOUBLEINDENT = INDENT + INDENT;
+        out.println(identifier);
+        out.println(INDENT + "coreConcurrency = " + coreConcurrency);
+        out.println(INDENT + "maxConcurrency = " + maxConcurrency);
+        out.println(INDENT + "maxQueueSize = " + maxQueueSize);
+        out.println(INDENT + "maxWaitForEnqueue = " + maxWaitForEnqueueNS.get() + " ms");
+        out.println(INDENT + "queueFullAction = " + queueFullAction.toString());
+        int numRunningThreads = maxConcurrency - maxConcurrencyConstraint.availablePermits();
+        int numRunningPrioritizedThreads = coreConcurrency - coreConcurrencyAvailable.get();
```

Comment:
Need to obtain the configLock when reading coreConcurrency/maxConcurrency values.  Should be able to do both computations within a single synchronized.